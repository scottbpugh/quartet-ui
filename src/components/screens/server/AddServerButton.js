// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import React, {Component} from "react";
import {Menu, MenuItem, Popover, Position} from "@blueprintjs/core";
import classNames from "classnames";
import {FormattedMessage} from "react-intl";

export class AddServerButton extends Component {
  goTo(path) {
    this.props.history.push(path);
  }

  render() {
    const addMenu = (
      <Menu
        className={classNames({
          "menu-padding-fix": true
        })}
      >
        <MenuItem
          text={<FormattedMessage id="app.serverSettings.addAServer" />}
          onClick={this.goTo.bind(this, "/server-settings/")}
        />
        <MenuItem
          text={<FormattedMessage id="app.plugins.addPlugin" />}
          onClick={this.goTo.bind(this, "/plugins")}
        />
      </Menu>
    );
    return (
      <div>
        <Popover
          className={classNames({"pt-dark": this.props.isDark})}
          content={addMenu}
          position={Position.RIGHT_CENTER}
        >
          <button
            onClick={this.displayMenu}
            tabIndex="0"
            className="pt-button pt-icon-add"
          >
            {/*
              <FormattedMessage
                id="plugins.numberRange.addServer"
                defaultMessage="Add a New Server"
              /> */}
          </button>
        </Popover>
      </div>
    );
  }
}
