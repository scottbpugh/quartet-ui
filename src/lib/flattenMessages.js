// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

export function flattenMessages(nestedMessages, prefix = "") {
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key];
    const prefixedKey = prefix ? `${prefix}.${key}` : key;

    if (typeof value === "string") {
      messages[prefixedKey] = value;
    } else {
      Object.assign(messages, flattenMessages(value, prefixedKey));
    }

    return messages;
  }, {});
}

window.qu4rtet.exports("lib/flattenMessages", this);
