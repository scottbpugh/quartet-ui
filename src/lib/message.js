// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {Position, Toaster, Intent} from "@blueprintjs/core";
import {pluginRegistry} from "plugins/pluginRegistration";
import stringHash from "string-hash";

const MINIMUM_EXPIRE = 7000;

const msgToaster = Toaster.create({
  className: "my-toaster",
  position: Position.BOTTOM_RIGHT
});

const getIntent = type => {
  switch (type) {
    case "success":
      return Intent.SUCCESS;
    case "danger":
    case "error":
      return Intent.DANGER;
    case "warning":
      return Intent.WARNING;
    default:
      return Intent.PRIMARY;
  }
};

const _msgHashMap = {};

const showMessageNoRepeat = (toastMessage, expires_in = MINIMUM_EXPIRE) => {
  let msgHash = stringHash(JSON.stringify(toastMessage));
  let timestamp = new Date().getTime();
  if (msgHash in _msgHashMap && _msgHashMap[msgHash].expires > timestamp) {
    // don't show msg and update the expiration to prevent the same message
    // over and over.
    _msgHashMap[msgHash].expires = timestamp + expires_in;
    return;
  }
  _msgHashMap[msgHash] = {expires: timestamp + expires_in};
  msgToaster.show(toastMessage);
};

export const showMessage = msg => {
  try {
    // for errors etc out of a component context.
    const intl = pluginRegistry.getIntl();
    if (msg.id) {
      showMessageNoRepeat(
        {
          message: intl.formatMessage({id: msg.id}, msg.values),
          intent: getIntent(msg.type)
        },
        msg.expires_in || MINIMUM_EXPIRE
      );
    } else if (msg.msg) {
      //let formatMsg = typeof msg.msg === "object" ? JSON.stringify(msg.msg) : msg.msg;
      showMessageNoRepeat(
        {message: msg.msg, intent: getIntent(msg.type)},
        msg.expires_in || MINIMUM_EXPIRE
      );
    }
  } catch (e) {
    console.log(
      "an error occurred while attempting to display an error toast.",
      e
    );
  }
};

window.qu4rtet.exports("lib/message", this);
