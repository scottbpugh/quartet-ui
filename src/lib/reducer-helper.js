// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

// A helper function meant to make life a little easier in reducers.

/**
 * Unknown - Description
 *
 * @param {object} [state={}]       The reducer state
 * @param {type}   serverID         The serverID
 * @param {object} [mergeObject={}] The object to merge into the state for this server.
 *
 * @return {type} Description
 */
export const setServerState = (state = {}, serverID, mergeObject = {}) => {
  if (!serverID) {
    throw new Error(
      "A server ID must be defined to set the server state in reducer."
    );
  }
  const copy = {...state}; // use a copy of state for manipulations.
  if (!copy.servers) {
    // if servers are not created yet, create an object.
    copy.servers = {};
  }
  if (!copy.servers[serverID]) {
    // if there is no object for this server, create it.
    copy.servers[serverID] = {};
  }
  return {
    ...copy,
    servers: {
      ...copy.servers,
      [serverID]: {
        ...copy.servers[serverID],
        ...mergeObject
      }
    }
  };
};
window.qu4rtet.exports("lib/reducer-helper", this);
