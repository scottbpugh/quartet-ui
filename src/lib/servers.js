// Copyright (c) 2018 Serial Lab
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import Swagger from "swagger-client";
import {pluginRegistry} from "plugins/pluginRegistration";
import {store} from "store";
import actions from "actions/serversettings";

/**
 * Server - Holds data about a server and its settings, API client, ...
 */
export class Server {
  constructor(serverSettings) {
    /*
    Following object should be passed.
    {
      serverSettingName,
      hostname,
      port,
      ssl,
      serverID,
      path,
      username,
      password
    }
    */
    this.setServerData(serverSettings);
    // make saved server object available to core and plugins.
    pluginRegistry.registerServer(this);
  }

  setServerData = serverSettings => {
    /*
    Following object should be passed.
    {
      serverSettingName,
      hostname,
      port,
      ssl,
      serverID,
      path,
      username,
      password
    */
    // allow for partial updates.
    for (let key in serverSettings) {
      if (serverSettings[key] !== "" || serverSettings[key] !== undefined) {
        this[key] = serverSettings[key];
      }
    }
    this.protocol = this.ssl === true ? "https" : "http";
    this.port = !this.port ? 80 : this.port;
    this.path = !this.path && !serverSettings.path ? "" : this.path;
    this.url = this.getServerURL();
    this._client = null;
    this.manifest = null;
    this.appList = [];
  };
  getFormStructure = () => {
    return Server.getFormStructure(this.toJSON());
  };
  static getFormStructure = (initialValues = {}) => [
    {
      name: "serverSettingName",
      defaultValue: initialValues.serverSettingName || null,
      description: {
        type: "text",
        required: true,
        read_only: false,
        label: "Server Setting Name",
        help_text:
          "The label that will be used for this server connection setting."
      }
    },
    {
      name: "serverID",
      defaultValue: initialValues.serverID || null,
      description: {
        type: "hidden",
        required: false,
        read_only: false,
        label: "Server ID",
        help_text: "Hidden Server ID"
      }
    },
    {
      name: "hostname",
      defaultValue: initialValues.hostname || null,
      description: {
        type: "text",
        required: true,
        read_only: false,
        label: "Server Hostname",
        help_text:
          "A hostname or IP address, example localhost, serial-box.com, or 192.168.5.10."
      }
    },
    {
      name: "port",
      defaultValue: initialValues.port || null,
      description: {
        type: "number",
        required: true,
        read_only: false,
        label: "Port Number",
        min_value: 1,
        max_value: 65535,
        help_text: "A port to connect to. Example, 80, 8080, 443, ..."
      }
    },
    {
      name: "path",
      defaultValue: initialValues.path || "",
      description: {
        type: "text",
        required: false,
        read_only: false,
        label: "Root Path",
        helperText: "A path to interact with API (Optional), example /api"
      }
    },
    {
      name: "ssl",
      defaultValue: initialValues.ssl || false,
      description: {
        type: "boolean",
        required: false,
        read_only: false,
        label: "SSL/TLS",
        helperText: "SSL/TLS encryption"
      }
    },
    {
      name: "username",
      defaultValue: initialValues.username || null,
      description: {
        type: "text",
        required: false,
        read_only: false,
        label: "Username",
        help_text: "Basic Auth Username"
      }
    },
    {
      name: "password",
      defaultValue: initialValues.password || null,
      description: {
        type: "password",
        required: false,
        read_only: false,
        label: "Password",
        help_text: "Basic Auth Password"
      }
    }
  ];
  getArrayFields = () => {
    return [
      {name: "protocol", value: this.protocol, editable: true},
      {name: "port", value: this.port, editable: true},
      {name: "path", value: this.path, editable: true},
      {name: "ssl", value: this.ssl, editable: true},
      {name: "url", value: this.url, editable: false},
      {name: "hostname", value: this.hostname, editable: true},
      {
        name: "serverSettingsName",
        value: this.serverSettingName,
        editable: true
      },
      {name: "username", value: this.username},
      {name: "serverID", value: this.serverID}
    ];
  };

  toJSON() {
    return {
      serverID: this.serverID,
      protocol: this.protocol,
      port: this.port,
      path: this.path,
      ssl: this.ssl,
      hostname: this.hostname,
      serverSettingName: this.serverSettingName,
      url: this.url,
      appList: this.appList,
      username: this.username,
      password: this.password
    };
  }

  getServerURL = () => {
    return `${this.protocol}://${this.hostname}:${this.port}/${this.path}`;
  };

  parseSchema = () => {
    return new Promise((resolve, reject) => {
      Swagger(this.url + "schema/")
        .then(client => {
          // swagger-js client is available.
          // client.spec / client.originalSpec / client.errors
          resolve(client);
        })
        .catch(error => {
          reject(error);
        });
    });
  };

  listApps = () => {
    if (this.appList.length === 0) {
      this.getClient()
        .then(client => {
          this.appList = Object.keys(client.apis);
          // let redux know we got our data
          store.dispatch({
            type: actions.appsListUpdated,
            payload: this.toJSON()
          });
        })
        .catch(error => {
          // handle legacy.
          this.appList = ["serialbox"];
          // let redux know we got our data
          store.dispatch({
            type: actions.appsListUpdated,
            payload: this.toJSON()
          });
        });
    }
  };

  getClient = () => {
    return new Promise((resolve, reject) => {
      if (this._client) {
        resolve(this._client);
      } else {
        // we don't have a client yet.
        // Fetch it.
        this.parseSchema()
          .then(client => {
            this._client = client;
            resolve(this._client);
          })
          .catch(error => {
            reject(error);
          });
      }
    });
  };

  getManifest = () => {
    new Promise((resolve, reject) => {
      if (!this.manifest) {
        this.getClient().then(client => {
          client.apis.manifest.quartet_manifest_list().then(result => {
            this.manifest = result.body;
            resolve(result.body);
          });
        });
      } else {
        resolve(this.manifest);
      }
    });
  };
}
