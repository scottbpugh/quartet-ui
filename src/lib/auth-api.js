// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
import {showMessage} from "lib/message";
import {getSyncValidators} from "components/elements/forms";
import {pluginRegistry} from "plugins/pluginRegistration";

/**
 * prepHeaders - Prepares the headers to be sent.
 *
 * @param {object} server A server setting object
 *
 * @return {object} A request init object with headers.
 */
export const prepHeaders = (server, method = "GET") => {
  let headers = new Headers();
  headers.append("Accept", "application/json");
  headers.append("Content-Type", "application/json");
  return {
    method: method,
    headers: headers,
    mode: "cors"
  };
};

/**
 * prepHeaders - Prepares the headers to be sent.
 *
 * @param {object} server A server setting object
 *
 * @return {object} A request init object with headers and basic auth.
 */
export const prepHeadersAuth = async (server, method = "GET") => {
  let headers = new Headers();
  headers.append("Accept", "application/json");
  headers.append("Content-Type", "application/json");
  headers.append(
    "Authorization",
    await pluginRegistry.getServer(server.serverID).getAuthorization()
  );
  return {
    method: method,
    headers: headers,
    mode: "cors"
  };
};

/**
 * getFormInfo - Description
 *
 * @param {A server object} server       Description
 * @param {Path to endpoint} path         Description
 * @param {Callback to set the state with formStructure} createForm   Description
 * @param {type} processField Description
 *
 */
export const getFormInfo = async (server, path, createForm, processField) => {
  return fetch(`${server.url}${path}`, await prepHeadersAuth(server, "OPTIONS"))
    .then(resp => {
      return resp.json();
    })
    .then(data => {
      // parse the values and filter to the one that are not readonly.
      let postFields = data.actions.POST;
      let formStructure = Object.keys(postFields)
        .map(field => {
          if (postFields[field].read_only === false) {
            return {name: field, description: postFields[field]};
          } else {
            return null;
          }
        })
        .filter(fieldObj => {
          if (processField) {
            return processField(fieldObj);
          }
          if (fieldObj) {
            // create sync validation arrays.
            fieldObj.validate = getSyncValidators(fieldObj);
            return true;
          }
          return false;
        });
      createForm(formStructure);
    })
    .catch(error => {
      showMessage({
        type: "danger",
        id: "app.servers.errorFormFetch",
        values: {error: error, serverName: server.serverSettingName}
      });
      throw error;
    });
};
