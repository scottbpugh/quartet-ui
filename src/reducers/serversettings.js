// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {handleActions} from "redux-actions";
import actions from "actions/serversettings";
import {showMessage} from "lib/message";
import uuidv4 from "uuid/v4";
import {pluginRegistry} from "plugins/pluginRegistration";
import {Server} from "lib/servers";

/**
 * initialData - Returns the initial data for
 * the Redux store for this form.
 * Used in index.js to initialized the store.
 *
 * It is a JSON-serializable object representing the form
 */
export const initialData = () => {
  let _initialData = {
    servers: {}, // uuid key to objects
    currentServer: null
  };
  return _initialData;
};

export const saveServer = postData => {
  return dispatch => {
    showMessage({type: "success", msg: "Your server settings were saved."});
    let server = {...postData};
    server.plugins = [];
    server.plugins.push("number-range");
    if (!postData.serverID) {
      postData.serverID = uuidv4();
    }
    dispatch({type: actions.saveServerSettings, payload: postData});
    const newServer = new Server(postData);
    pluginRegistry.registerServer(newServer);
    return dispatch({
      type: actions.serverUpdated,
      payload: JSON.stringify(newServer)
    });
  };
};

export const loadCurrentServer = serverData => {
  return dispatch => {
    return dispatch({type: actions.loadCurrentServer, payload: serverData});
  };
};

export default handleActions(
  {
    [actions.loadCurrentServer]: (state, action) => {
      return {
        ...state,
        formData: initialData(action.payload).formData
      };
    },
    [actions.saveServerSettings]: (state, action) => {
      return {
        ...state,
        servers: {...state.servers, [action.payload.serverID]: action.payload},
        formData: initialData().formData
      };
    },
    [actions.appsListUpdated]: (state, action) => {
      return {
        ...state,
        servers: {...state.servers, [action.payload.serverID]: action.payload}
      };
    }
  },
  {}
);
