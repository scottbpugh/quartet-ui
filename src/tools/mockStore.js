// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React from "react";

import {IntlProvider} from "react-intl";
import {addLocaleData} from "react-intl";
import messages from "messages";
import en from "react-intl/locale-data/en";
import fr from "react-intl/locale-data/fr";
import {flattenMessages} from "lib/flattenMessages";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

addLocaleData([...en, ...fr]);
let defaultLocale = "en-US";
const middlewares = [thunk];

export const initialState = {
  dashboard: {notifications: []},
  serversettings: {
    servers: {
      "824ae2dd-2411-4adf-a846-d9b310a33d4b": {
        serverSettingName: "Server #1",
        serverName: "localhost",
        port: "80",
        ssl: false,
        username: "root",
        password: "toor",
        serverID: "824ae2dd-2411-4adf-a846-d9b310a33d4b"
      }
    }
  },
  intl: {
    defaultLocale: defaultLocale,
    locale: defaultLocale,
    messages: flattenMessages(messages[defaultLocale])
  },
  layout: {pageTitle: {id: "app.nav.servers"}},
  plugins: {plugins: {}, navTreeItems: []}
};

export const mockStore = configureStore(middlewares);

export const TestWrapper = ({locale, messages, children}) => {
  if (locale === undefined) {
    locale = defaultLocale;
  }
  if (messages === undefined) {
    messages = initialState.intl.messages;
  }
  return (
    <IntlProvider locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
};

