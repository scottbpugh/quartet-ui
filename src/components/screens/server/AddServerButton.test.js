// Copyright (c) 2018 SerialLab Corp.
//
// GNU GENERAL PUBLIC LICENSE
//    Version 3, 29 June 2007
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import React, {Component} from "react";
import renderer from "react-test-renderer";
import {AddServerButton} from "./AddServerButton";
import {initialData} from "../../../reducers/serversettings";
import configureStore from "redux-mock-store";
import {MemoryRouter as Router, withRouter} from "react-router-dom";
import {IntlProvider, intlReducer} from "react-intl-redux";
import {Provider} from "react-redux";
import {mockStore, TestWrapper, initialState} from "tools/mockStore";
import {Server} from "lib/servers";
import {FormattedMessage} from "react-intl";

it("renders correctly", () => {
  let store = mockStore(initialState);

  const serverButton = renderer
    .create(
      <TestWrapper>
        <Provider store={store}>
          <Router>
            <AddServerButton />
          </Router>
        </Provider>
      </TestWrapper>
    )
    .toJSON();
  expect(serverButton).toMatchSnapshot();
});

