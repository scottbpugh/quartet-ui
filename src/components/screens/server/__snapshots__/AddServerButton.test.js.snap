// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <span
      className="bp3-popover-target"
      onClick={[Function]}
    >
      <button
        className="bp3-button add"
      />
    </span>
  </div>
</div>
`;
